### PR TITLE
Changed the directory Structure to MVT Model and merge New Menubar

### PR DESCRIPTION
**Pull request by fatimarafiqui**
I have converted the directory structure to the one proposed on this: [MISC] Propose and Create the directory structure for the complete application
The composer.json is a dependancy manager that handles all packages eg. twilio. composer.lock is responsible of locking the last composer update.
- pagetop.php and pagebottom.php are separated to make the code modular
- pagetop has everything that was required in the header + the file that needed to be imported.
- pagebottom includes js files.
- menu.html is for rendering the menu. we have loops for each item of menu and the for every sub-menu item.
- menu.php makes use of twig for rendering
- We have settings.php that stores all credentials for two modes: deployment mode [for Heroku] and debug mode [local host] and uses credentials on its own based on the mode it detects.
- index.php - It is the home page responsible for rendering all the pages. It is short and modular.
Covered the following tasks:
https://trello.com/c/atvfo9xb
https://trello.com/c/jL8qAQis
https://trello.com/c/gAdslDKH

@AkankshaBodhankar @MehaKaushik @npari 

new_menu

### CODE REVIEW

**Comment by npari** [comment]
@fatimarafiqui I am seeing conflicts in ReadMe.md

**Comment by fatimarafiqui** [comment]
@npari Resolved the merge conflict. It was nothing, just a word change by @AkankshaBodhankar that created the conflict.
conflict

**Comment by AkankshaBodhankar** [comment]
@fatimarafiqui As many files are changed it would be good if you create different commits. Reviewing 63,000+ changes is difficult.

File: includes/application.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+	$APPLICATION_DIR = str_replace("includes","", dirname(__FILE__));
+
+	spl_autoload_register(function ($class_name) {
+		global $APPLICATION_DIR;
+	    include $APPLICATION_DIR.'/modules/'.$class_name . '.php';
+	});
+	
+	require_once($APPLICATION_DIR.'/includes/settings.php');
```
Comment (prathmeshranaut) [inline includes/application.php]: Add this in the `composer autoloader`
Check PSR-0 or PSR-4 file inclusion using autoloader for this.

File: includes/settings.php [hunk 2]
```diff
@@ -5,13 +5,33 @@
...
-$_settings['twilio']['auth_token'] = '';
-$_settings['twilio']['number'] = '';
+$_settings['twilio']['account_sid'] = getenv('account_sid') ?  getenv('account_sid') : '';
+$_settings['twilio']['auth_token'] = getenv('auth_token') ? getenv('auth_token') : '';
+$_settings['twilio']['number'] = getenv('number') ?  getenv('number') : '';
+
+//'heroku' for production, 'debug' for running locally
+$_settings['deploy_mode'] = strpos($_SERVER['REQUEST_URI'], 'herokuapp.com') ? 'heroku' : 'debug';
 
+if ($_settings['deploy_mode'] == 'heroku') {
```
Comment (prathmeshranaut) [inline includes/settings.php]: You shouldn't tie the database down to just one deployment target. Use something like
```
if ($development) {
    //Set the DB details
} else {
    //Fetch the DB details from env variable
}
```

Or even better you can get rid of the if block and always load the configuration from the env file.

File: index.php [hunk 3]
```diff
@@ -1,18 +1,8 @@
...
-</head>
-<body>
-<center>
-    <img src="images/PClogoWhite.png" style="width:200px; height:200px;float:left;"/>
-    <h1 class="title">First Aide</h1>
-</center>
-</body>
 <?php
-    header("refresh:2;url=login.php");//redirect to login automatically after refreshing 2 times
+    require_once(dirname(__FILE__).'/includes/page_top.php');
```
Comment (prathmeshranaut) [inline index.php]: Why are you using ```dirname(__FILE__)``` here and `$APPLICATION_DIR` in the next line?

Comment (prathmeshranaut) [inline index.php]: Indent Line 2 and 3 properly as well

File: javascripts/menu.js [hunk 4]
```diff
@@ -1,41 +1,9 @@
...
-		$("#supportServices").next().slideDown();
-	}	
-
-	if(Cookies.get("sexualAssaultAwareness")=='1'){
-		$("#sexualAssaultAwareness").next().slideDown();
+$('.sub-menu-header').click(function() {
+	var thisParent = $(this).parent();
+	var thisMenu = thisParent.find('.menu');
+	if (thisMenu.is(':hidden')) {
+		thisParent.find('.menu').slideDown( "slow");
```
Comment (prathmeshranaut) [inline javascripts/menu.js]: Use variable `thisMenu` here and in the else statement

File: modules/Authentication.php [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+		function __construct($email, $password) {
+			if (!($email && $password)) {
+				return false;
+			}
+			echo "<div>In Authentication constructor</div>";
+			$user = new User();
+		}
+	}
+?>
```
Comment (prathmeshranaut) [inline modules/Authentication.php]: Remove the closing tag when you are creating a pure PHP only file.

File: modules/Authentication.php [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?php
+	class Authentication {
+		private $user;
+
+		function __construct($email, $password) {
+			if (!($email && $password)) {
+				return false;
+			}
+			echo "<div>In Authentication constructor</div>";
```
Comment (prathmeshranaut) [inline modules/Authentication.php]: Remove this.

File: modules/Router.php [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	class Router {
+		const HOME = 'index';
+		const DASHBOARD = 'dashboard';
+		
+		public static function getPage($page) {
+			$out = array(
+				'found' => true
+			);
+			switch($page) {
+				case self::DASHBOARD:	$out['type'] = self::DASHBOARD;
```
Comment (prathmeshranaut) [inline modules/Router.php]: Indentation issue and put these 2 statements in two lines.

File: modules/User.php [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?php
+	class User {
+		function __construct() {
+			echo "<div>In User constructor</div>";
```
Comment (prathmeshranaut) [inline modules/User.php]: Remove this

**Comment by prathmeshranaut** [comment]
Compare each point on http://www.php-fig.org/psr/psr-2/ with your existing code and make sure you follow each and every rule.

**Comment by fatimarafiqui** [comment]
@prathmeshranaut I have made sure to use PSR-2 rules at each and every line of code, even for the parenthesis.

**Comment by prathmeshranaut** [comment]
@fatimarafiqui Are you using spaces? It looks from github that its tabs. Just confirming.

**Comment by fatimarafiqui** [comment]
@prathmeshranaut I have used for spaces for each indentation.

**Comment by prathmeshranaut** [comment]
Awesome.

File: template/index.html [hunk 9]
```diff
@@ -0,0 +1 @@
+Index page
```
Comment (fatimarafiqui) [inline template/index.html]: Right. It will be reflected in the next PR.

File: index.php [hunk 10]
```diff
@@ -1,18 +1,8 @@
...
-</center>
-</body>
 <?php
-    header("refresh:2;url=login.php");//redirect to login automatically after refreshing 2 times
-?>
-</html>
\ No newline at end of file
+	$APPLICATION_DIR = dirname(__FILE__);
+	require_once($APPLICATION_DIR.'/includes/page_top.php');
+	require_once($APPLICATION_DIR.'/includes/menu.php');
```
Comment (fatimarafiqui) [inline index.php]: @AkankshaBodhankar Yes, I have already added the splash screen. It will be reflected in the next PR.

**Comment by fatimarafiqui** [comment]
@AkankshaBodhankar Made the suggested changes. Please verify and merge the PR.

File: includes/page_top.php [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+	$page = Router::getPage($query);
+	$javascripts = array(
+		'jquery-3.2.1.min.js',
+		'semantic.min.js'
+	);
+?>
+<html>
+<head>
+
+<!-- Standard Meta -->
```
Comment (AkankshaBodhankar) [inline includes/page_top.php]: Have you done this ?

Comment (AkankshaBodhankar) [inline includes/page_top.php]: Have you done this ?

Comment (fatimarafiqui) [inline includes/page_top.php]: Yes https://github.com/systers/FirstAide-web/pull/231/commits/19dbd51ca5514931e9a14a53e6ec5a3709b8db01

Comment (fatimarafiqui) [inline includes/page_top.php]: Yes. https://github.com/systers/FirstAide-web/pull/231/commits/2174f1ee664fb11fcb15359ee707a2e403cb00db

**Comment by AkankshaBodhankar** [comment]
@fatimarafiqui From the next PR onwards try to use proper commit messages and also try not to make commits for small changes. Once you have submitted the PR it is not a good practice to create new commits daily until it is merged. If you feel anything is left out you can discuss about it first and then create a commit or include it in the new PR.

**Comment by AkankshaBodhankar** [comment]
This PR is good to go.

**Pull request merged**